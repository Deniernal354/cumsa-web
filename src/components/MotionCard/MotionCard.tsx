import { motion } from "framer-motion";
import { ReactNode } from "react";

type MotionCardProps = {
  children: ReactNode;
  className?: string;
};

export default function MotionCard({ children, className = "" }: MotionCardProps) {
  return (
    <motion.div
      whileHover={{ y: -4, boxShadow: "0 8px 32px 0 rgba(0,0,0,0.10)" }}
      transition={{ type: "spring", stiffness: 300, damping: 24 }}
      className={"rounded-xl bg-white ring-1 ring-black/10 shadow-sm p-6 " + className}
    >
      {children}
    </motion.div>
  );
}
