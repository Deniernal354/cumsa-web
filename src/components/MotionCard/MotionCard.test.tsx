import { render } from "@testing-library/react";
import MotionCard from "./MotionCard";

test("MotionCard snapshot", () => {
  const { asFragment } = render(<MotionCard>Test</MotionCard>);
  expect(asFragment()).toMatchSnapshot();
});
