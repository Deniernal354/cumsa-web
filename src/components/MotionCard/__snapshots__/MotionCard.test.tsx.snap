// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MotionCard snapshot 1`] = `
<DocumentFragment>
  <div
    class="rounded-xl bg-white ring-1 ring-black/10 shadow-sm p-6 "
  >
    Test
  </div>
</DocumentFragment>
`;
